core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -85,6 +85,9 @@ private[spark] class Executor(
     env.blockManager.initialize(conf.getAppId)
   }
 
+  private val executorMetrics: ExecutorMetrics = new ExecutorMetrics
+  executorMetrics.setHostname(Utils.localHostName)
```

Comment:
It's better to use HOST:PORT, but executor port cannot get here, which should get from `RpcEnv.adress.port`. We can get `executorId` on the driver side when receiving the message, that might be enough to identify the different executors, but since we will show the removed executors on the page, so we cannot know where the executor locate by only `executorId`, because the Executor tab only shows the active executors. we can remove the hostname here if we support showing the removed executors on Executor tab.